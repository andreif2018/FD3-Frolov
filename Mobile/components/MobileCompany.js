import React from 'react';
import PropTypes from 'prop-types';

import MobileClient from './MobileClient';

import './MobileCompany.css';

class MobileCompany extends React.PureComponent {

    static propTypes = {
        name: PropTypes.string.isRequired,
        headers: PropTypes.arrayOf(
            PropTypes.shape({
                code: PropTypes.number.isRequired,
                header: PropTypes.string.isRequired,
            })
        ),
        clients:PropTypes.arrayOf(
            PropTypes.shape({
                id: PropTypes.number.isRequired,
                lastName: PropTypes.string.isRequired,
                firstName: PropTypes.string.isRequired,
                otchestvo: PropTypes.string.isRequired,
                balance: PropTypes.number.isRequired,
                status: PropTypes.bool.isRequired,
            })
        ),
    };

    state = {
        name: this.props.name,
        clients: this.props.clients,
        showState: 0 // show all
    };

    setName1 = () => {
        this.setState({name:'МТС'});
    };

    setName2 = () => {
        this.setState({name:'Velcom'});
    };

    setBalance = (clientId,newBalance) => {
      let changed=false;
      let newClients=[...this.state.clients]; // копия самого массива клиентов
      newClients.forEach( (c,i) => {
        if ( c.id===clientId && c.balance!==newBalance ) {
          let newClient={...c}; // копия хэша изменившегося клиента
          newClient.balance=newBalance;
          newClients[i]=newClient;
          changed=true;
        }
      } );
      if ( changed )
        this.setState({clients:newClients});
    };

    showAll = () => {
        this.setState({showState: 0} ); // show all
    };

    showActive = () => {
        this.setState({showState: 1} ); // show active
    };

    showBlocked = () => {
        this.setState({showState: 2} ); // show blocked
    };

    addClient = () => {

    };

    render() {

        console.log("MobileCompany render");

        var headerLine = this.props.headers.map((element) => <th key={element.code}>{element.header}</th>);/* формирование заголовков таблицы */

        var clientsCode=this.state.clients.map( client =>
            <MobileClient key={client.id} info={client} show={this.state.showState} />
        );


        return (
            <div className='MobileCompany'>
                <input type="button" value="МТС" onClick={this.setName1} />
                <input type="button" value="Velcom" onClick={this.setName2} />
                <div className='MobileCompanyName'>Компания &laquo;{this.state.name}&raquo;</div>
                <hr/>
                <input type="button" value="Все" onClick={this.showAll} />
                <input type="button" value="Активные" onClick={this.showActive} />
                <input type="button" value="Заблокированные" onClick={this.showBlocked} />
                <hr/>
                <table className="MobileCompanyClients">
                    <thead><tr>{headerLine}</tr></thead>
                    <tbody>{clientsCode}</tbody>
                </table>
                <input type="button" value="Добавить клиента" onClick={this.addClient} />
            </div>
        )
            ;

    }

}

export default MobileCompany;