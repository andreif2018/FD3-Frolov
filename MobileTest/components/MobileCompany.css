.MobileCompany {
    width: auto;
    border: 2px solid lightgray;
    padding: 10px;
    text-align: left;
}

input[type='button'] {
    margin: 20px 0 20px 0 ;
}

td, th {
    width: 140px;
    height: 2em;
    border-top: 1px solid darkgrey;
}