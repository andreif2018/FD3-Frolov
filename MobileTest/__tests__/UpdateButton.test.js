"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import MobileCompany from '../components/MobileCompany';
import Enzyme, {mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

let clientsArr = [
    {id:101, lastName:"Иванов", firstName: "Иван", otchestvo: "Иванович", balance: 200, status: true},
    {id:102, lastName:"Сидоров", firstName: "Сидор", otchestvo: "Сидорович", balance: 250, status: true},
    {id:103, lastName:"Петров", firstName: "Петр", otchestvo: "Петрович", balance: 180, status: true},
    {id:104, lastName:"Григорьев", firstName: "Григорий", otchestvo: "Григорьевич", balance: -220, status: false},
];
let headers = [
    {code:1, header: "Фамилия"},
    {code:2, header: "Имя"},
    {code:3, header: "Отчество"},
    {code:4, header: "Баланс"},
    {code:5, header: "Статус"},
    {code:6, header: "Редактировать"},
    {code:7, header: "Удалить"},
];

test('нажатие кнопки "Редактировать" в каждой строчке таблицы', () => {

    for (var buttonIndex = 0; buttonIndex < 4; buttonIndex++) {
        const component = renderer.create(
            <MobileCompany headers = {headers} clients={clientsArr}/>
        );
        let componentTree=component.toJSON();
        // найдём в вёрстке компонента саму кнопку
        const buttonElem = component.root.findAll( el => (el.type==='input' && el.props.value === 'Редактировать') )[buttonIndex];
        // и "нажмём" на неё
        buttonElem.props.onClick();
        // получаем уже изменённый снэпшот
        componentTree=component.toJSON();
        expect(componentTree).toMatchSnapshot('UpdateButton.test' + buttonIndex + '.js.snap');// режим редактирования
    }
});

it('check input value', () => {
    let inputLastName = "Changed";
    const wrapper = mount(<MobileCompany headers = {headers} clients={clientsArr}/>);
    wrapper.find({ value: 'Редактировать'}).first().simulate('click');
    expect(wrapper).toMatchSnapshot("editMode");
    wrapper.find({defaultValue: 'Иванов'}).props().value = inputLastName;
    expect(wrapper).toMatchSnapshot("UpdatedLastName");
    console.log(wrapper.find({ value: 'Сохранить'}).value);
    wrapper.find({ value: 'Сохранить'}).simulate('click');
    expect(wrapper).toMatchSnapshot("SavedUpdatedLastName");


    // !!! не верно сохраняет данные
    console.log("не верно сохраняет данные");

});

