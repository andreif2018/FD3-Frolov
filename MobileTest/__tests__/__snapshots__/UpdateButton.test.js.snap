// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedUpdatedLastName 1`] = `
<MobileCompany
  clients={
    Array [
      Object {
        "balance": 200,
        "firstName": "Иван",
        "id": 101,
        "lastName": "Иванов",
        "otchestvo": "Иванович",
        "status": true,
      },
      Object {
        "balance": 250,
        "firstName": "Сидор",
        "id": 102,
        "lastName": "Сидоров",
        "otchestvo": "Сидорович",
        "status": true,
      },
      Object {
        "balance": 180,
        "firstName": "Петр",
        "id": 103,
        "lastName": "Петров",
        "otchestvo": "Петрович",
        "status": true,
      },
      Object {
        "balance": -220,
        "firstName": "Григорий",
        "id": 104,
        "lastName": "Григорьев",
        "otchestvo": "Григорьевич",
        "status": false,
      },
    ]
  }
  headers={
    Array [
      Object {
        "code": 1,
        "header": "Фамилия",
      },
      Object {
        "code": 2,
        "header": "Имя",
      },
      Object {
        "code": 3,
        "header": "Отчество",
      },
      Object {
        "code": 4,
        "header": "Баланс",
      },
      Object {
        "code": 5,
        "header": "Статус",
      },
      Object {
        "code": 6,
        "header": "Редактировать",
      },
      Object {
        "code": 7,
        "header": "Удалить",
      },
    ]
  }
>
  <div
    className="MobileCompany"
  >
    <input
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
    <hr />
    <table
      className="MobileCompanyClients"
    >
      <thead>
        <tr>
          <th
            key="1"
          >
            Фамилия
          </th>
          <th
            key="2"
          >
            Имя
          </th>
          <th
            key="3"
          >
            Отчество
          </th>
          <th
            key="4"
          >
            Баланс
          </th>
          <th
            key="5"
          >
            Статус
          </th>
          <th
            key="6"
          >
            Редактировать
          </th>
          <th
            key="7"
          >
            Удалить
          </th>
        </tr>
      </thead>
      <tbody>
        <MobileClient
          info={
            Object {
              "balance": 200,
              "firstName": "Иван",
              "id": 101,
              "lastName": "Иванов",
              "otchestvo": "Иванович",
              "status": true,
            }
          }
          key="101"
        >
          <tr>
            <td>
              Иванов
            </td>
            <td>
              Иван
            </td>
            <td>
              Иванович
            </td>
            <td>
              200
            </td>
            <td
              className="Active"
            >
              active
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Редактировать"
              />
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Удалить"
              />
            </td>
          </tr>
        </MobileClient>
        <MobileClient
          info={
            Object {
              "balance": 250,
              "firstName": "Сидор",
              "id": 102,
              "lastName": "Сидоров",
              "otchestvo": "Сидорович",
              "status": true,
            }
          }
          key="102"
        >
          <tr>
            <td>
              Сидоров
            </td>
            <td>
              Сидор
            </td>
            <td>
              Сидорович
            </td>
            <td>
              250
            </td>
            <td
              className="Active"
            >
              active
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Редактировать"
              />
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Удалить"
              />
            </td>
          </tr>
        </MobileClient>
        <MobileClient
          info={
            Object {
              "balance": 180,
              "firstName": "Петр",
              "id": 103,
              "lastName": "Петров",
              "otchestvo": "Петрович",
              "status": true,
            }
          }
          key="103"
        >
          <tr>
            <td>
              Петров
            </td>
            <td>
              Петр
            </td>
            <td>
              Петрович
            </td>
            <td>
              180
            </td>
            <td
              className="Active"
            >
              active
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Редактировать"
              />
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Удалить"
              />
            </td>
          </tr>
        </MobileClient>
        <MobileClient
          info={
            Object {
              "balance": -220,
              "firstName": "Григорий",
              "id": 104,
              "lastName": "Григорьев",
              "otchestvo": "Григорьевич",
              "status": false,
            }
          }
          key="104"
        >
          <tr>
            <td>
              Григорьев
            </td>
            <td>
              Григорий
            </td>
            <td>
              Григорьевич
            </td>
            <td>
              -220
            </td>
            <td
              className="Blocked"
            >
              blocked
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Редактировать"
              />
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Удалить"
              />
            </td>
          </tr>
        </MobileClient>
      </tbody>
    </table>
    <input
      onClick={[Function]}
      type="button"
      value="Добавить клиента"
    />
  </div>
</MobileCompany>
`;

exports[`UpdateButton.test0.js.snap 1`] = `
<div
  className="MobileCompany"
>
  <input
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="MobileCompanyClients"
  >
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <input
            defaultValue="Иванов"
            maxLength="20"
            minLength="2"
            required={true}
            type="text"
          />
        </td>
        <td>
          <input
            defaultValue="Иван"
            maxLength="20"
            minLength="2"
            required={true}
            type="text"
          />
        </td>
        <td>
          <input
            defaultValue="Иванович"
            maxLength="20"
            minLength="2"
            required={true}
            type="text"
          />
        </td>
        <td>
          <input
            className="Balance"
            defaultValue={200}
            max="10000"
            min="-1000"
            required={true}
            type="number"
          />
        </td>
        <td
          className="Active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Сохранить"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="Active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Петров
        </td>
        <td>
          Петр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="Active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="Blocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`UpdateButton.test1.js.snap 1`] = `
<div
  className="MobileCompany"
>
  <input
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="MobileCompanyClients"
  >
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="Active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          <input
            defaultValue="Сидоров"
            maxLength="20"
            minLength="2"
            required={true}
            type="text"
          />
        </td>
        <td>
          <input
            defaultValue="Сидор"
            maxLength="20"
            minLength="2"
            required={true}
            type="text"
          />
        </td>
        <td>
          <input
            defaultValue="Сидорович"
            maxLength="20"
            minLength="2"
            required={true}
            type="text"
          />
        </td>
        <td>
          <input
            className="Balance"
            defaultValue={250}
            max="10000"
            min="-1000"
            required={true}
            type="number"
          />
        </td>
        <td
          className="Active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Сохранить"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Петров
        </td>
        <td>
          Петр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="Active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="Blocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`UpdateButton.test2.js.snap 1`] = `
<div
  className="MobileCompany"
>
  <input
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="MobileCompanyClients"
  >
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="Active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="Active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          <input
            defaultValue="Петров"
            maxLength="20"
            minLength="2"
            required={true}
            type="text"
          />
        </td>
        <td>
          <input
            defaultValue="Петр"
            maxLength="20"
            minLength="2"
            required={true}
            type="text"
          />
        </td>
        <td>
          <input
            defaultValue="Петрович"
            maxLength="20"
            minLength="2"
            required={true}
            type="text"
          />
        </td>
        <td>
          <input
            className="Balance"
            defaultValue={180}
            max="10000"
            min="-1000"
            required={true}
            type="number"
          />
        </td>
        <td
          className="Active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Сохранить"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="Blocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`UpdateButton.test3.js.snap 1`] = `
<div
  className="MobileCompany"
>
  <input
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="MobileCompanyClients"
  >
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="Active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="Active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Петров
        </td>
        <td>
          Петр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="Active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          <input
            defaultValue="Григорьев"
            maxLength="20"
            minLength="2"
            required={true}
            type="text"
          />
        </td>
        <td>
          <input
            defaultValue="Григорий"
            maxLength="20"
            minLength="2"
            required={true}
            type="text"
          />
        </td>
        <td>
          <input
            defaultValue="Григорьевич"
            maxLength="20"
            minLength="2"
            required={true}
            type="text"
          />
        </td>
        <td>
          <input
            className="Balance"
            defaultValue={-220}
            max="10000"
            min="-1000"
            required={true}
            type="number"
          />
        </td>
        <td
          className="Blocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Сохранить"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`UpdatedLastName 1`] = `
<MobileCompany
  clients={
    Array [
      Object {
        "balance": 200,
        "firstName": "Иван",
        "id": 101,
        "lastName": "Иванов",
        "otchestvo": "Иванович",
        "status": true,
      },
      Object {
        "balance": 250,
        "firstName": "Сидор",
        "id": 102,
        "lastName": "Сидоров",
        "otchestvo": "Сидорович",
        "status": true,
      },
      Object {
        "balance": 180,
        "firstName": "Петр",
        "id": 103,
        "lastName": "Петров",
        "otchestvo": "Петрович",
        "status": true,
      },
      Object {
        "balance": -220,
        "firstName": "Григорий",
        "id": 104,
        "lastName": "Григорьев",
        "otchestvo": "Григорьевич",
        "status": false,
      },
    ]
  }
  headers={
    Array [
      Object {
        "code": 1,
        "header": "Фамилия",
      },
      Object {
        "code": 2,
        "header": "Имя",
      },
      Object {
        "code": 3,
        "header": "Отчество",
      },
      Object {
        "code": 4,
        "header": "Баланс",
      },
      Object {
        "code": 5,
        "header": "Статус",
      },
      Object {
        "code": 6,
        "header": "Редактировать",
      },
      Object {
        "code": 7,
        "header": "Удалить",
      },
    ]
  }
>
  <div
    className="MobileCompany"
  >
    <input
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
    <hr />
    <table
      className="MobileCompanyClients"
    >
      <thead>
        <tr>
          <th
            key="1"
          >
            Фамилия
          </th>
          <th
            key="2"
          >
            Имя
          </th>
          <th
            key="3"
          >
            Отчество
          </th>
          <th
            key="4"
          >
            Баланс
          </th>
          <th
            key="5"
          >
            Статус
          </th>
          <th
            key="6"
          >
            Редактировать
          </th>
          <th
            key="7"
          >
            Удалить
          </th>
        </tr>
      </thead>
      <tbody>
        <MobileClient
          info={
            Object {
              "balance": 200,
              "firstName": "Иван",
              "id": 101,
              "lastName": "Иванов",
              "otchestvo": "Иванович",
              "status": true,
            }
          }
          key="101"
        >
          <tr>
            <td>
              <input
                defaultValue="Иванов"
                maxLength="20"
                minLength="2"
                required={true}
                type="text"
                value="Changed"
              />
            </td>
            <td>
              <input
                defaultValue="Иван"
                maxLength="20"
                minLength="2"
                required={true}
                type="text"
              />
            </td>
            <td>
              <input
                defaultValue="Иванович"
                maxLength="20"
                minLength="2"
                required={true}
                type="text"
              />
            </td>
            <td>
              <input
                className="Balance"
                defaultValue={200}
                max="10000"
                min="-1000"
                required={true}
                type="number"
              />
            </td>
            <td
              className="Active"
            >
              active
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Сохранить"
              />
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Удалить"
              />
            </td>
          </tr>
        </MobileClient>
        <MobileClient
          info={
            Object {
              "balance": 250,
              "firstName": "Сидор",
              "id": 102,
              "lastName": "Сидоров",
              "otchestvo": "Сидорович",
              "status": true,
            }
          }
          key="102"
        >
          <tr>
            <td>
              Сидоров
            </td>
            <td>
              Сидор
            </td>
            <td>
              Сидорович
            </td>
            <td>
              250
            </td>
            <td
              className="Active"
            >
              active
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Редактировать"
              />
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Удалить"
              />
            </td>
          </tr>
        </MobileClient>
        <MobileClient
          info={
            Object {
              "balance": 180,
              "firstName": "Петр",
              "id": 103,
              "lastName": "Петров",
              "otchestvo": "Петрович",
              "status": true,
            }
          }
          key="103"
        >
          <tr>
            <td>
              Петров
            </td>
            <td>
              Петр
            </td>
            <td>
              Петрович
            </td>
            <td>
              180
            </td>
            <td
              className="Active"
            >
              active
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Редактировать"
              />
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Удалить"
              />
            </td>
          </tr>
        </MobileClient>
        <MobileClient
          info={
            Object {
              "balance": -220,
              "firstName": "Григорий",
              "id": 104,
              "lastName": "Григорьев",
              "otchestvo": "Григорьевич",
              "status": false,
            }
          }
          key="104"
        >
          <tr>
            <td>
              Григорьев
            </td>
            <td>
              Григорий
            </td>
            <td>
              Григорьевич
            </td>
            <td>
              -220
            </td>
            <td
              className="Blocked"
            >
              blocked
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Редактировать"
              />
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Удалить"
              />
            </td>
          </tr>
        </MobileClient>
      </tbody>
    </table>
    <input
      onClick={[Function]}
      type="button"
      value="Добавить клиента"
    />
  </div>
</MobileCompany>
`;

exports[`editMode 1`] = `
<MobileCompany
  clients={
    Array [
      Object {
        "balance": 200,
        "firstName": "Иван",
        "id": 101,
        "lastName": "Иванов",
        "otchestvo": "Иванович",
        "status": true,
      },
      Object {
        "balance": 250,
        "firstName": "Сидор",
        "id": 102,
        "lastName": "Сидоров",
        "otchestvo": "Сидорович",
        "status": true,
      },
      Object {
        "balance": 180,
        "firstName": "Петр",
        "id": 103,
        "lastName": "Петров",
        "otchestvo": "Петрович",
        "status": true,
      },
      Object {
        "balance": -220,
        "firstName": "Григорий",
        "id": 104,
        "lastName": "Григорьев",
        "otchestvo": "Григорьевич",
        "status": false,
      },
    ]
  }
  headers={
    Array [
      Object {
        "code": 1,
        "header": "Фамилия",
      },
      Object {
        "code": 2,
        "header": "Имя",
      },
      Object {
        "code": 3,
        "header": "Отчество",
      },
      Object {
        "code": 4,
        "header": "Баланс",
      },
      Object {
        "code": 5,
        "header": "Статус",
      },
      Object {
        "code": 6,
        "header": "Редактировать",
      },
      Object {
        "code": 7,
        "header": "Удалить",
      },
    ]
  }
>
  <div
    className="MobileCompany"
  >
    <input
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
    <hr />
    <table
      className="MobileCompanyClients"
    >
      <thead>
        <tr>
          <th
            key="1"
          >
            Фамилия
          </th>
          <th
            key="2"
          >
            Имя
          </th>
          <th
            key="3"
          >
            Отчество
          </th>
          <th
            key="4"
          >
            Баланс
          </th>
          <th
            key="5"
          >
            Статус
          </th>
          <th
            key="6"
          >
            Редактировать
          </th>
          <th
            key="7"
          >
            Удалить
          </th>
        </tr>
      </thead>
      <tbody>
        <MobileClient
          info={
            Object {
              "balance": 200,
              "firstName": "Иван",
              "id": 101,
              "lastName": "Иванов",
              "otchestvo": "Иванович",
              "status": true,
            }
          }
          key="101"
        >
          <tr>
            <td>
              <input
                defaultValue="Иванов"
                maxLength="20"
                minLength="2"
                required={true}
                type="text"
              />
            </td>
            <td>
              <input
                defaultValue="Иван"
                maxLength="20"
                minLength="2"
                required={true}
                type="text"
              />
            </td>
            <td>
              <input
                defaultValue="Иванович"
                maxLength="20"
                minLength="2"
                required={true}
                type="text"
              />
            </td>
            <td>
              <input
                className="Balance"
                defaultValue={200}
                max="10000"
                min="-1000"
                required={true}
                type="number"
              />
            </td>
            <td
              className="Active"
            >
              active
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Сохранить"
              />
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Удалить"
              />
            </td>
          </tr>
        </MobileClient>
        <MobileClient
          info={
            Object {
              "balance": 250,
              "firstName": "Сидор",
              "id": 102,
              "lastName": "Сидоров",
              "otchestvo": "Сидорович",
              "status": true,
            }
          }
          key="102"
        >
          <tr>
            <td>
              Сидоров
            </td>
            <td>
              Сидор
            </td>
            <td>
              Сидорович
            </td>
            <td>
              250
            </td>
            <td
              className="Active"
            >
              active
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Редактировать"
              />
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Удалить"
              />
            </td>
          </tr>
        </MobileClient>
        <MobileClient
          info={
            Object {
              "balance": 180,
              "firstName": "Петр",
              "id": 103,
              "lastName": "Петров",
              "otchestvo": "Петрович",
              "status": true,
            }
          }
          key="103"
        >
          <tr>
            <td>
              Петров
            </td>
            <td>
              Петр
            </td>
            <td>
              Петрович
            </td>
            <td>
              180
            </td>
            <td
              className="Active"
            >
              active
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Редактировать"
              />
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Удалить"
              />
            </td>
          </tr>
        </MobileClient>
        <MobileClient
          info={
            Object {
              "balance": -220,
              "firstName": "Григорий",
              "id": 104,
              "lastName": "Григорьев",
              "otchestvo": "Григорьевич",
              "status": false,
            }
          }
          key="104"
        >
          <tr>
            <td>
              Григорьев
            </td>
            <td>
              Григорий
            </td>
            <td>
              Григорьевич
            </td>
            <td>
              -220
            </td>
            <td
              className="Blocked"
            >
              blocked
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Редактировать"
              />
            </td>
            <td>
              <input
                onClick={[Function]}
                type="button"
                value="Удалить"
              />
            </td>
          </tr>
        </MobileClient>
      </tbody>
    </table>
    <input
      onClick={[Function]}
      type="button"
      value="Добавить клиента"
    />
  </div>
</MobileCompany>
`;
